ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -138,8 +138,8 @@ void ClientResize(HWND hWnd, int nWidth, int nHeight)
   MoveWindow(hWnd, rcWind.left, rcWind.top, nWidth + ptDiff.x, nHeight + ptDiff.y, TRUE);
 }
 
-WCHAR wstrPromptStr[4096];
-WCHAR wstrTextEntry[MAX_PATH];
+WCHAR wstrPromptStr[4096 + 1];
+WCHAR wstrTextEntry[MAX_PATH + 1];
```

Comment:
Instead of repeating this change about 38 times, you could have just `#define` a new constant at the top, in case you need to change it again. Also `4096 + 1` is a bit of a nono, people use just `4096` because it's an even power of two and can be a lot faster if it fits into a single memory page.

Look at this and notice that 4 KiB is exactly 4096 bytes.
https://en.wikipedia.org/wiki/Page_(computer_memory)
![image](https://user-images.githubusercontent.com/3212801/44553231-cf9ce180-a6fa-11e8-8ae9-7c4f38812a74.png)